From c57c77e528433af6b7e889c10caa9faa44b1b68e Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Tue, 21 May 2024 19:33:17 +0200
Subject: [PATCH] Build mozangle with TARGET_CXXFLAGS

Make sure to use TARGET_CXXFLAGS when compiling mozangle,
to ensure that it doesn't use the system's own headers,
and that it can be actually cross-compiled.
---
 build.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/bitbake/mozangle-0.5.1/build.rs b/bitbake/mozangle-0.5.1/build.rs
index 97ea2be..df6b51a 100644
--- a/bitbake/mozangle-0.5.1/build.rs
+++ b/bitbake/mozangle-0.5.1/build.rs
@@ -349,6 +349,12 @@ fn build_translator(compiled_libraries: &mut HashSet<Libs>, target: &String) {
         .clang_arg("c++")
         .clang_arg("-std=c++17");
 
+    if let Ok(flags) = env::var("TARGET_CXXFLAGS") {
+        for flag in flags.split_whitespace() {
+            builder = builder.clang_arg(flag);
+        }
+    }
+
     for func in ALLOWLIST_FN {
         builder = builder.allowlist_function(func)
     }
-- 
2.45.0

